core/shared/src/main/scala/zio/prelude/NonEmptySet.scala

```diff
@@ -83,6 +95,11 @@ final class NonEmptySet[A] private (private val set: Set[A]) { self =>
   /** Removes the `elem` from this `NonEmptySet`. Alias for `-`. */
   def remove(elem: A): Set[A] = set - elem
 
+  /**
+   * Returns the tail of this `NonEmptySet` if it exists or `None` otherwise.
+   */
+  def tailOption: Option[NonEmptySet[A]] = destructEither.toOption.map(_._2)
```

Comment:
Not sure these make sense for a set that is not ordered. This seems to imply that there is a "head" and "tail" which is not the case.

Reply:
True, there is no order. But sometimes you need to be able to access _some_ element from the collection, even though there isn't any order and the element is picked "arbitrarily". And sometimes you may need to have access to the rest of the collection without that element. And sometimes you may need both.